### ISSUE TITLE
query vaults with prefix containig special characters fails

### ISSUE THREAD

**Issue by simplecoin-eu** [fireblocks/java-sdk#14]
they produce invalid signature because uri should be urlencoded

**Comment by PavelNiedoba** [comment]
it's been 2 weeks without any activity

**Comment by asafs932** [comment]
Hi, can you please share a bit of information on which method and query fields you encountered it?
was it "getPagedVaultAccounts" with "namePrefix" query field?

**Comment by PavelNiedoba** [comment]
This fails when vaultname contains ':' or any character which gets urlEncoded to something else. It's kotlin code but you get an idea.

```
  private fun loadAllCompanyVaults(company: String) {
        companyVaults.clear()
        val vaults = workspace.getFireblocks().vaults().getPagedVaultAccounts(
            "$company:",
            null,
            null,
            null,
            "ASC",
            null,
            null,
            BigDecimal.valueOf(100),
        ).get().data.accounts

        vaults.forEach { vault ->
            companyVaults[vault.name] = vault
        }

        log.info("Loaded ${companyVaults.size} company vaults")
    }
```
screenshot is showing param names
![image](https://github.com/fireblocks/java-sdk/assets/7063851/6a9e5de8-bf00-4c7d-95cc-33142537ee29)

**Comment by PavelNiedoba** [comment]
Also I'm almost sure that it's problem of any parameter passed in url

**Comment by asafs932** [comment]
I was able to reproduce it, it occurs with query input which includes special characters as you mentioned.
Thanks for bringing that up, we are on it and will update soon with the fix.

**Comment by asafs932** [comment]
Issue was found and a fix is currently under code review

**Comment by asafs932** [comment]
@PavelNiedoba issue is solved in the new version v2.1.0
